x-pack/solutions/security/plugins/security_solution/scripts/telemetry/build_ebt_data_view.ts

```diff
@@ -146,40 +146,89 @@ function removeTrailingSlash(url: string) {
     return url;
   }
 }
+interface NestedSchemaNode {
+  type?: string;
+  properties?: NestedObject;
+  items?: NestedSchemaNode;
+  [key: string]: unknown;
+}
 interface NestedObject {
-  [key: string]: { type?: string; properties?: NestedObject };
+  [key: string]: NestedSchemaNode;
+}
+
+interface InspectSchemaNodeResult {
+  childToInspect?: NestedObject;
+  nodeType?: string;
+}
+
+function isObjectRecord(x: unknown): x is Record<string, unknown> {
+  return typeof x === 'object' && x !== null;
+}
```

Comment:
nit: `isObjectRecord` might be removed in favor of  Lodash's [isObject](https://lodash.com/docs/4.17.15#isObject).